nmdc_automation/re_iding/scripts/re_id_tool.py

```diff
@@ -92,6 +91,188 @@ def cli(ctx, target):
+        metabolomics_activity_records = db_client["metabolomics_analysis_activity_set"].find(
+            {"was_informed_by": omics_legacy_id}
+        )
+        len_metabolomics_activity_records = len(list(metabolomics_activity_records.clone()))
+        logging.info(f"Found {len_metabolomics_activity_records} Metabolomics Activity records for {omics_legacy_id}")
+
+
+        for metabolomics_activity_record in metabolomics_activity_records:
+            metabolomics_legacy_id = metabolomics_activity_record["id"]
+            logging.info(f"Updating Metabolomics Activity record: {metabolomics_legacy_id}")
+            # get the data objects for has_input and has_output
+            has_input = metabolomics_activity_record.get("has_input", [])
+            has_output = metabolomics_activity_record.get("has_output", [])
+            input_data_objects = db_client["data_object_set"].find({"id": {"$in": has_input}})
+            logging.info(f"Found {len(list(input_data_objects.clone()))} input data objects for {metabolomics_legacy_id}")
+            output_data_objects = db_client["data_object_set"].find({"id": {"$in": has_output}})
+            logging.info(f"Found {len(list(output_data_objects.clone()))} output data objects for {metabolomics_legacy_id}")
+
+            # Mint new IDs if we are updating the database
+            if not no_update:
+                with session.start_transaction():
+                    try:
+                        # update the has_input data objects and the parent OmicsProcessing record
+                        new_input_data_object_ids = _update_metabolomics_input_data_objects(
+                            input_data_objects, db_client, api_client, updated_record_identifiers
+                            )
+                        # update has_output in the parent OmicsProcessing record
+                        op_filter_criteria = {"id": omics_new_id}
+                        op_update_criteria = {"$set": {"has_output": new_input_data_object_ids}}
+                        result = db_client["omics_processing_set"].update_one(op_filter_criteria, op_update_criteria)
+                        logging.info(f"Updated {result.modified_count} omics_processing_set records")
+
+                        # update the has_output data objects and the Metabolomics Activity record
+                        metabolomics_new_id = api_client.minter("nmdc:MetabolomicsAnalysisActivity")
+                        non_orphan_metabolomics_ids.append(metabolomics_new_id)
+                        updated_record_identifiers.append(("metabolomics_analysis_activity_set", metabolomics_legacy_id, metabolomics_new_id))
+                        new_output_data_object_ids = _update_metabolomics_output_data_objects(
+                            output_data_objects, metabolomics_new_id, db_client, api_client, updated_record_identifiers
+                            )
+                        # update the Metabolomics Activity record
```

Comment:
This block is missing updating has_input for the metabolomics activity record.

Reply:
https://github.com/microbiomedata/nmdc_automation/pull/116/commits/5066cbf06d9156a0b0b598c246e4f43a6ed0ad2b